rust/hyperlane-base/src/db/typed_db.rs

```diff
@@ -57,8 +76,7 @@ impl TypedDB {
         key: &K,
         value: &V,
     ) -> Result<()> {
-        self.db
-            .store_keyed_encodable(self.full_prefix(prefix), key, value)
+        self.store_encodable(prefix, key.to_vec(), value)
```

Comment:
Should this function be public? Feels like a potential footgun to have public functions that don't handle the key prefixing

Reply:
Especially since full_key is private rn

Reply:
This function requires the prefix as an argument. I don't understand your comment.

Reply:
Tbh coming back to this I don't think I do either. My recollection was that some public functions took a prefixed key, and others did not, but it looks like every public fn takes a prefix and a key